src/createServer.js

```diff
@@ -1,13 +1,230 @@
+/* eslint-disable no-console */
 'use strict';
 
-// const express = require('express');
+const express = require('express');
+// const { v4: uuidv4 } = require('uuid');
+const cors = require('cors');
+// const { query } = require('express');
 
 function createServer() {
-  // Use express to create a server
-  // Add a routes to the server
-  // Return the server (express app)
+  const app = express();
+  let users = [];
+  let expenses = [];
+
+  app.use(express.json({ extended: true }));
+  app.use(cors());
+
+  app.get('/users', (req, res) => {
+    if (!users.length) {
+      res.send([]);
+    } else {
+      res.statusCode = 200;
+      res.send(users);
+    }
+  });
+
+  app.get('/users/:userId', (req, res) => {
+    const id = parseInt(req.path.slice(1).split('/')[1]);
+
+    const userItem = users.find(user => user.id === id);
+    const isUser = users.some(user => user.id === id);
+
+    if (isUser) {
+      res.statusCode = 200;
+      res.send(userItem);
+    } else {
+      res.setHeader('Content-Type', /application\/json/);
+      res.sendStatus(404);
+      res.end();
+    }
+  });
+
+  app.post('/users', (req, res) => {
+    const { name } = req.body;
+    const newUser = {
+      id: 1,
+      name,
+    };
+
+    if (!name) {
+      res.statusCode = 400;
+      res.end();
+      users = [];
+    } else {
+      res.statusCode = 201;
+      users.push(newUser);
+      res.send(newUser);
+      console.log(users);
+
+      return newUser;
+    }
+  });
+
+  app.patch('/users/:userId', (req, res) => {
+    const newName = req.body.name;
+    const id = parseInt(req.path.slice(1).split('/')[1]);
+
+    let myUser = {};
+
+    users.map((user) => {
+      if (user.id === id) {
+        Object.assign(user, { 'name': newName });
+        myUser = user;
+      } else {
+        res.setHeader('Content-Type', /application\/json/);
+      }
+    });
+    res.send(myUser);
+  });
+
+  app.delete('/users/:userId', (req, res) => {
+    const id = parseInt(req.path.slice(1).split('/')[1]);
+
+    res.statusCode = 404;
+
+    users.map((user, index) => {
+      if (user.id === id) {
+        users.splice(index, 1);
+        res.statusCode = 204;
+      }
+    });
+
```

Comment:
`map` is not the best method in this case, use `forEach` and `break` the loop after the `if` is done